lib/model.js

```diff
@@ -194,11 +189,24 @@ var Events = require('./events'),
             this._isDestructed = true;
             this.trigger('destruct');
             this._eventEmitter.removeAllListeners();
-            Object.keys(this.attributes).forEach(function (attributeName) {
-                this.attributes[attributeName].destruct();    
-            }, this);
+            this.eachAttribute(function (attribute) {
+                attribute.destruct();
+            });
         },
 
+        /**
+         * @param {Function} cb
+         * @param {Object} [ctx]
+         */
+        eachAttribute: function (cb, ctx) {
+            this._attributesNames.forEach(function (attrName) {
```

Comment:
I suppose it could be enough to use `this` as a context. In opposite, the `cb` could be bound to another context with a `bind`.